### PR TITLE
feat: Added support for response headers policy

### LINKED ISSUE #56: Add Resource: aws_cloudfront_response_headers_policy
## Is your request related to a new offering from AWS?
Amazon CloudFront introduces Response Headers Policies
https://aws.amazon.com/blogs/networking-and-content-delivery/amazon-cloudfront-introduces-response-headers-policies/

## Describe the solution you'd like.
Remove the need to create `Lambda @Edge` for CORS control, since it is now available out of the box

## Additional context
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_response_headers_policy

### PR DESCRIPTION
**Pull request by hinupurthakur**
## Description

Adds the support for Response Headers Policy
## Motivation and Context

The requirement for the same in mentioned in the issue #56 

Fixes #56 
## How Has This Been Tested?
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -89,7 +89,7 @@ module "cloudfront" {
   default_cache_behavior = {
     target_origin_id       = "appsync"
     viewer_protocol_policy = "allow-all"
-
+    response_headers_policy_id = "67f7725c-6f97-4210-82d7-5512b31e9d03"
```
Comment (antonbabenko) [inline examples/complete/main.tf:92]: Add a comment next to this `magic id` saying that this Policy ID was taken from https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-response-headers-policies.html

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Please update `versions.tf` to reflect the minimum version of Terraform AWS provider where this feature was introduced. Also, fix formatting and update the docs by running `pre-commit run -a`.

File: examples/complete/main.tf (hunk 1 above)
Comment (hinupurthakur) [reply examples/complete/main.tf]: @antonbabenko Please check now.

**Comment by antonbabenko** [comment]
Thanks, @hinupurthakur !

[v2.9.0](https://github.com/terraform-aws-modules/terraform-aws-cloudfront/releases/v2.9.0) has been just released.

**Pull request merged**